lib/accumulate_distribute/util/gen_order_amounts.js

```diff
@@ -27,14 +28,14 @@ const generateOrderAmounts = (args = {}) => {
     while (Math.abs(amount - totalAmount) > DUST) {
       const m = Math.random() > 0.5 ? 1 : -1
       const orderAmount = Math.min(sliceAmount, sliceAmount * (1 + (Math.random() * amountDistortion * m)))
-      const remAmount = amount - totalAmount
+      const remAmount = nBN(amount).minus(totalAmount).toNumber()
       const cappedOrderAmount = +prepareAmount(remAmount < 0
         ? Math.max(remAmount, orderAmount)
         : Math.min(remAmount, orderAmount)
       )
 
       orderAmounts.push(cappedOrderAmount)
-      totalAmount += cappedOrderAmount
+      totalAmount = nBN(totalAmount).plus(cappedOrderAmount).toNumber()
```

Comment:
ok, something is wrong here. if i revert all changes here in the file and run the testsuite, tests still pass for me.